README.md

```diff
@@ -3,156 +3,351 @@
+operations that would otherwise fill up the user's storage device.
 
-## Background
 
-Quota is the system which sets and enforces limits on disk usage by the browser and web applications. Up until recently (M80), the quota system would factor in free disk space when determining how much quota each origin is allowed. This allowed applications to rely on the quota system to know exactly how much disk space they had to write to. **Without information about what will and will not fit on disk, applications that depend on storage usage cannot guarantee a working experience to their users.**
+## Background
 
-In the current design, free disk is no longer incorporated into the calculation of an application’s storage quota due to a security issue. This leaves applications and developers in a place where they have no guarantees around how much space they can write to, and must wait for write errors before realizing they need to scale back their offline caching.
+The Storage standard
+[recommends](https://storage.spec.whatwg.org/#storage-quota) that
+"[the storage quota of a storage shelf] should be less than the total
+available storage space on the device to give users some wiggle room." In the
+past, Chrome treated this recommendation as a requirement, and used the free
+disk space as a hard cap for quota. In other words, for certain users, a
+site's quota would match the available disk space.
+
+Chrome's approach became [a security problem](https://crbug.com/960305) when
+combined with the
+[API exposed in the Storage Standard](https://storage.spec.whatwg.org/#api),
+which includes
+[a quota estimate method](https://storage.spec.whatwg.org/#dom-storagemanager-estimate).
+The method allows sites to obtain their available quota which, for some Chrome
+users, used to reflect their free disk space. The ability to precisely
+measure free disk space is a security problem, because it can be used to measure
+the size of cross-origin resources. The same ability may also be used for
+fingerprinting.
+
+In order to fix the security problem described above, Chrome switched to
+granting sites a fixed amount of quota, which does not account for the amount of
+free disk space.
+
+While this change fixed the security problem, it was a regression for web
+applications that relied on the details of Chrome's implementation of the
+Storage API to provide useful functionality to their users.
+
+One class of regressions is avoidig downloading a large corpus of data when
+the free disk space is low. For example, some Google properties would not
+enable offline sync for users with low free disk space. This was an attempt
+to avoid a bad user experience -- enabling offline sync would lead to a large
+amount of data being downloaded and stored locally, which would lead to
+running out of disk space, which could cause system instability such as OS
+crashes. This proposal aims to address this class of regressions.
```

Comment:
Consider bold for "This proposal aims to address..."

Reply:
Done. Thank you!